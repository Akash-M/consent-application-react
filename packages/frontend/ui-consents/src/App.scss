@import "lib-components/src/scss/main";

main {
  align-items: flex-start;
  display: flex;
  flex-flow: column nowrap;
  height: 100vh;
  width: 100%;
}
