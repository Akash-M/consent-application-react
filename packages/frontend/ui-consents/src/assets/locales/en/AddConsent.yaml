header: Give consent
username: Name
email: Email Address
disclaimer: 'I agree to:'
addButton: Give consent
success: Consent added successfully
errors:
  username: Please enter a valid name
  email: Please enter a valid email id
  consents: Please ensure that you agree to at least one consent
  api: Error fetching consent list. Please try again in some time.
