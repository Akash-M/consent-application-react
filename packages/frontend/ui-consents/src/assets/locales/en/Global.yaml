headers:
  addConsent: Give consent
  listConsent: Collected consents
consents:
  newsletter: Receive newsletter
  ads: Be shown targeted ads
  statistics: Contribute to anonymous visit statistics
loader: Loading...
