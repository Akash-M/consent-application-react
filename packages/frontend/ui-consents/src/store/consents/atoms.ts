import { atom } from 'recoil';

export const ConsentListState = atom<Consent.Detail[]>({
  key: 'ConsentListState',
  default: [],
});

export const ConsentListPaginatorState = atom<Consent.Paginator>({
  key: 'ConsentListPaginatorState',
  default: {
    currentPage: 1,
    perPage: 2,
  },
});
