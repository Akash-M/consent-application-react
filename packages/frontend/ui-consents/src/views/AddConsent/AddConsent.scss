.add-consent {
  display: flex;
  flex-direction: column;
  margin: 1rem;

  @include tablet {
    width: auto;
  }

  @include desktop {
    margin: 0;
    max-width: 1024px;
    width: auto;
  }

  &__user-info {
    display: flex;
    flex-direction: column;
    margin-bottom: 2rem;

    .text-field {
      height: 40px;
    }

    .text-field:first-child {
      margin-bottom: 1rem;
    }

    @include tablet {
      flex-direction: row;

      .text-field {
        min-width: 250px;
      }

      .text-field:first-child {
        margin-bottom: 0;
      }

      .text-field:last-child {
        margin-left: 1rem;
      }
    }
  }

  &__consents {
    display: flex;
    flex-direction: column;
    margin-bottom: 2rem;
  }

  > button {
    align-self: flex-start;
    width: fit-content;
  }
}
