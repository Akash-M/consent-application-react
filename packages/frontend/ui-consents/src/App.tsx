import React from 'react';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

import '$/App.scss';
import SiteHeader from '$/components/SiteHeader';
import Layout from '$/containers/Layout';

const App = (): JSX.Element => {
  return (
    <main>
      <SiteHeader />

      <Layout />

      <ToastContainer />
    </main>
  );
};

export default App;
