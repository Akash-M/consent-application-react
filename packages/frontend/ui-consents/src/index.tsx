import React, { Suspense } from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { RecoilRoot } from 'recoil';

import setI18n from 'lib-utils/src/i18n';

import App from '$/App';

export function Root(): JSX.Element {
  return (
    <React.StrictMode>
      <RecoilRoot>
        <Router>
          <Suspense fallback={null}>
            <App />
          </Suspense>
        </Router>
      </RecoilRoot>
    </React.StrictMode>
  );
}

setI18n('Global', true, { en: '0.2' });

const rootElement = document.querySelector('#root');
ReactDOM.render(<Root />, rootElement);
