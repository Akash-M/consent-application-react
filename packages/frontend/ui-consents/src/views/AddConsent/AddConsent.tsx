import Button from '@mui/material/Button';
import Checkbox from '@mui/material/Checkbox';
import FormControl from '@mui/material/FormControl';
import FormControlLabel from '@mui/material/FormControlLabel';
import FormGroup from '@mui/material/FormGroup';
import FormHelperText from '@mui/material/FormHelperText';
import FormLabel from '@mui/material/FormLabel';
import TextField from '@mui/material/TextField';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { toast } from 'react-toastify';

import { postConsent } from 'lib-api/src/consent';
import { EMAIL_REGEX } from 'lib-utils/src/email-regex';

import './AddConsent.scss';

export function AddConsent(): JSX.Element {
  const { t } = useTranslation(['AddConsent']);

  const [formData, setFormData] = useState<Consent.NewEntry>({
    username: '',
    email: '',
    consent: {
      newsletter: false,
      ads: false,
      statistics: false,
    },
  });
  const [isFormValid, setIsFormValid] = useState(true);

  const areConsentsValid = () => {
    return (
      formData.consent.newsletter ||
      formData.consent.ads ||
      formData.consent.statistics
    );
  };

  const updateConsent = (event: React.BaseSyntheticEvent) => {
    setFormData({
      username: formData.username,
      email: formData.email,
      consent: {
        ...formData.consent,
        [event.target.name]: event.target.checked,
      },
    });
  };

  const validateForm = () => {
    if (
      formData.username.length > 0 &&
      EMAIL_REGEX.test(formData.email) &&
      areConsentsValid()
    ) {
      setIsFormValid(true);
    } else {
      setIsFormValid(false);
    }
  };

  const handleSubmit = async (e: React.BaseSyntheticEvent) => {
    e.preventDefault();
    /* istanbul ignore else */
    if (isFormValid) {
      try {
        await postConsent(formData);
        toast.success(t('AddConsent.success'), {
          position: toast.POSITION.BOTTOM_RIGHT,
        });
      } catch {
        toast.error(t('AddConsent.errors.api'), {
          position: toast.POSITION.BOTTOM_RIGHT,
        });
      }
    }
  };

  return (
    <article className="add-consent">
      <h2>{t('AddConsent.header')}</h2>

      <form noValidate onSubmit={handleSubmit}>
        <section className="add-consent__user-info">
          <TextField
            required
            className="text-field"
            defaultValue={formData.username}
            error={!isFormValid && formData.username.length === 0}
            label={t('AddConsent.username')}
            name="username"
            variant="standard"
            onChange={(e) =>
              setFormData({ ...formData, username: e.target.value })
            }
          />

          <TextField
            required
            className="text-field"
            defaultValue={formData.email}
            error={!isFormValid && !EMAIL_REGEX.test(formData.email)}
            label={t('AddConsent.email')}
            name="email"
            variant="standard"
            onChange={(e) =>
              setFormData({ ...formData, email: e.target.value })
            }
          />
        </section>

        <section className="add-consent__consents">
          <FormControl
            required
            component="fieldset"
            error={!isFormValid && !areConsentsValid()}
            variant="standard"
          >
            <FormLabel component="legend">
              {t('AddConsent.disclaimer')}
            </FormLabel>

            <FormGroup>
              <FormControlLabel
                control={
                  <Checkbox
                    checked={formData.consent.newsletter}
                    name="newsletter"
                    onChange={updateConsent}
                  />
                }
                label={t<string>('Global.consents.newsletter')}
              />

              <FormControlLabel
                control={
                  <Checkbox
                    checked={formData.consent.ads}
                    name="ads"
                    onChange={updateConsent}
                  />
                }
                label={t<string>('Global.consents.ads')}
              />

              <FormControlLabel
                control={
                  <Checkbox
                    checked={formData.consent.statistics}
                    name="statistics"
                    onChange={updateConsent}
                  />
                }
                label={t<string>('Global.consents.statistics')}
              />
            </FormGroup>

            {!isFormValid && (
              <FormHelperText>{t('AddConsent.errors.consents')}</FormHelperText>
            )}
          </FormControl>
        </section>

        <Button type="submit" variant="contained" onClick={validateForm}>
          {t('AddConsent.addButton')}
        </Button>
      </form>
    </article>
  );
}
