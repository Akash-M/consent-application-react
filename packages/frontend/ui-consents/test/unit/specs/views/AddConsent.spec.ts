import { fireEvent, screen, waitFor } from '@testing-library/react';
import { setI18n } from 'react-i18next';

import { postConsent } from 'lib-api/src/consent';
import { consentListFixtures } from 'lib-api/src/fixtures';
import { customRenderer } from 'lib-utils/src/testing/factory';
import { I18N_MISSING_KEY, loadI18n } from 'lib-utils/src/testing/i18n';

import App from '$/App';
import AddConsent from '$/assets/locales/en/AddConsent.yaml';
import Global from '$/assets/locales/en/Global.yaml';
import { ConsentListState } from '$/store/consents/atoms';

jest.mock('lib-api/src/consent', () => ({
  postConsent: jest.fn(),
}));

const initializeState = ({ set }: any) => {
  set(ConsentListState, consentListFixtures);
};

describe('<App />', () => {
  beforeAll(() => {
    setI18n(loadI18n('Global', { Global, AddConsent }));
  });

  beforeEach(jest.clearAllMocks);

  describe('should render add consent entry screen', () => {
    test('without any form validation messages by default', () => {
      const { container } = customRenderer(App, initializeState);
      expect(container.firstChild!.textContent).not.toContain(I18N_MISSING_KEY);
    });

    test('with form validation messages when user clicks on submit and form is invalid', async () => {
      const { container } = customRenderer(App, initializeState);
      fireEvent.click(screen.getByText('AddConsent.addButton'));
      await waitFor(() => {
        expect(screen.getByText('AddConsent.errors.consents')).toBeTruthy();
      });
      expect(container.firstChild!.textContent).not.toContain(I18N_MISSING_KEY);
    });
  });

  describe('should trigger api call when user submits form', () => {
    test('displays toast message on success', async () => {
      const mockEntry: Consent.NewEntry = {
        username: 'user6',
        email: 'user6@email.com',
        consent: {
          newsletter: true,
          ads: true,
          statistics: true,
        },
      };

      (postConsent as jest.Mock).mockResolvedValueOnce(mockEntry);
      customRenderer(App, initializeState);
      const inputs = screen.getAllByRole('textbox');
      fireEvent.change(inputs[0], { target: { value: mockEntry.username } });
      fireEvent.change(inputs[1], { target: { value: mockEntry.email } });
      fireEvent.click(screen.getByText('Global.consents.newsletter'));
      fireEvent.click(screen.getByText('Global.consents.ads'));
      fireEvent.click(screen.getByText('Global.consents.statistics'));
      fireEvent.click(screen.getByText('AddConsent.addButton'));
      await waitFor(() => {
        expect(postConsent).toHaveBeenCalledTimes(1);
      });
      await waitFor(() => {
        expect(screen.getByText('AddConsent.success')).toBeTruthy();
      });
    });

    test('displays toast message on error', async () => {
      (postConsent as jest.Mock).mockRejectedValueOnce('mock-error');
      customRenderer(App, initializeState);
      const inputs = screen.getAllByRole('textbox');

      fireEvent.change(inputs[0], { target: { value: 'user6' } });
      fireEvent.change(inputs[1], { target: { value: 'user6@email.com' } });
      fireEvent.click(screen.getByText('Global.consents.newsletter'));
      fireEvent.click(screen.getByText('Global.consents.ads'));
      fireEvent.click(screen.getByText('Global.consents.statistics'));
      fireEvent.click(screen.getByText('AddConsent.addButton'));
      await waitFor(() => {
        expect(postConsent).toHaveBeenCalledTimes(1);
      });
      await waitFor(() => {
        expect(screen.getByText('AddConsent.errors.api')).toBeTruthy();
      });
    });
  });
});
