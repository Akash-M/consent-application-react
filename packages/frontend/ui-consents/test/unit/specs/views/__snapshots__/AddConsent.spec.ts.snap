// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render add consent entry screen with form validation messages when user clicks on submit and form is invalid 1`] = `
<main>
  <nav
    class="site-header"
  >
    <h2>
      Consents App
    </h2>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-whjmaz-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="Brightness4Icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </nav>
  <article
    class="layout"
  >
    <aside
      class="layout__links"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Global.headers.addConsent
      </a>
      <a
        class=""
        href="/list-consent"
      >
        Global.headers.listConsent
      </a>
    </aside>
    <section
      class="layout__screens"
    >
      <article
        class="add-consent"
      >
        <h2>
          AddConsent.header
        </h2>
        <form
          novalidate=""
        >
          <section
            class="add-consent__user-info"
          >
            <div
              class="MuiFormControl-root MuiTextField-root text-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error Mui-required css-1petlo4-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                AddConsent.username
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk Mui-error css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-19ywia9-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="true"
                  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="username"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root text-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error Mui-required css-1petlo4-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                AddConsent.email
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk Mui-error css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-19ywia9-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="true"
                  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="email"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </section>
          <section
            class="add-consent__consents"
          >
            <fieldset
              class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
            >
              <legend
                class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error Mui-required css-o8ft5d-MuiFormLabel-root"
              >
                AddConsent.disclaimer
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk Mui-error css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </legend>
              <div
                class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-1gaxl8k-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      name="newsletter"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Global.consents.newsletter
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-1gaxl8k-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      name="ads"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Global.consents.ads
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-1gaxl8k-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      name="statistics"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Global.consents.statistics
                  </span>
                </label>
              </div>
              <p
                class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium Mui-required css-11apz7u-MuiFormHelperText-root"
              >
                AddConsent.errors.consents
              </p>
            </fieldset>
          </section>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-8h7phy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            AddConsent.addButton
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </article>
    </section>
  </article>
  <div
    class="Toastify"
  />
</main>
`;

exports[`<App /> should render add consent entry screen without any form validation messages by default 1`] = `
<main>
  <nav
    class="site-header"
  >
    <h2>
      Consents App
    </h2>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-whjmaz-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="Brightness4Icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </nav>
  <article
    class="layout"
  >
    <aside
      class="layout__links"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Global.headers.addConsent
      </a>
      <a
        class=""
        href="/list-consent"
      >
        Global.headers.listConsent
      </a>
    </aside>
    <section
      class="layout__screens"
    >
      <article
        class="add-consent"
      >
        <h2>
          AddConsent.header
        </h2>
        <form
          novalidate=""
        >
          <section
            class="add-consent__user-info"
          >
            <div
              class="MuiFormControl-root MuiTextField-root text-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1petlo4-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                AddConsent.username
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19ywia9-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="username"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root text-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1petlo4-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                AddConsent.email
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19ywia9-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="email"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </section>
          <section
            class="add-consent__consents"
          >
            <fieldset
              class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
            >
              <legend
                class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-o8ft5d-MuiFormLabel-root"
              >
                AddConsent.disclaimer
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </legend>
              <div
                class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-1gaxl8k-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      name="newsletter"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Global.consents.newsletter
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-1gaxl8k-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      name="ads"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Global.consents.ads
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-1gaxl8k-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      name="statistics"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Global.consents.statistics
                  </span>
                </label>
              </div>
            </fieldset>
          </section>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-8h7phy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            AddConsent.addButton
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </article>
    </section>
  </article>
  <div
    class="Toastify"
  />
</main>
`;
